pages/stacking-promotions/Checkout.tsx

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from "react";
+import { MetaProperties } from "../../components/MetaProperties/Meta";
+import styles from "../../styles/CartAndCheckout.module.css";
+import { getCartAndVoucherFromSessionStorage } from "../../utils/stacking-promotions/sessionStorage";
+import { EachProduct } from "../types";
+import { VouchersProperties } from "../types";
+import FormPreview from "../../components/stacking-promotions/FormPreview/FormPreview";
+import CheckoutSummary from "../../components/stacking-promotions/CheckoutSummary/CheckoutSummary";
+
+const Checkout = () => {
+  const [products, setProducts] = useState<EachProduct[]>([]);
+  const [vouchersProperties, setVouchersProperties] =
+    useState<VouchersProperties>();
+
+  useEffect(() => {
+    const { storageProducts: products, vouchersProperties } =
+      getCartAndVoucherFromSessionStorage();
+    setProducts(products);
+    setVouchersProperties(vouchersProperties);
+  }, []);
+  console.log(vouchersProperties);
+  return (
+    <>
+      <MetaProperties
+        title={"Stacking promotions"}
+        description={"Checkout view"}
+        keywords={"example, voucherify, cart shop"}
+      />
+      <div className={styles.checkoutPageWrapper}>
+        <FormPreview
+          products={products}
+          vouchersProperties={vouchersProperties as VouchersProperties}
```

Comment:
it's dangerous

```ts
vouchersProperties={vouchersProperties as VouchersProperties}
```

now you tell me why :)

Reply:
okay, i tell typescript the type VouchersProperties can be undefined too and then we don't need to tell strictly that it can be only as VouchersProperties